GA4 enhancement (#676)
rudderlabs/rudderstack-docs: docs/destinations/analytics/google-analytics-4/setting-up-google-analytics-4-in-rudderstack.mdx

    - **Firebase App Id**: Enter the Firebase App ID which is the identifier for Firebase app. It can be found in the Firebase dashboard under **Project Settings > General**.
 
<div class="infoBlock">
Refer to the <a href="#faq">FAQ</a> section for more information on how to obtain the API Secret, Measurement Id and Firebase App ID.
</div>

### Identify

<img src="../../../assets/event-stream-destinations/google-analytics-4-connection-settings-1.png" alt="Google Analytics 4 Connection Settings in RudderStack" />

> - **Enable Server Side Identify**: Enable this setting to fire an <Link to="/rudderstack-api/api-specification/rudderstack-spec/identify/">`identify`</Link> call.

Reviewer: ```suggestion
- **Enable Server Side Identify**: Enable this setting to make an <Link to="/rudderstack-api/api-specification/rudderstack-spec/identify/">`identify`</Link> call.
```

>   If enabled, you can set the following:

Reviewer: ```suggestion
If **Enable Server Side Identify** setting is enabled, you can also configure the following settings:
```

Reviewer: Clarified a bit - lmk if it makes sense.

> - **Trait to differentiate between new or existing user**: Enter a trait key to determine whether its a new or an existing user.

Reviewer: ```suggestion
- **Trait to differentiate between new or existing user**: Enter a trait key to determine whether a user is new or an existing user.
```

- **Convert identify event to Login or Signup event**: Enable this setting to convert the `identify` event to a `login` event in case of an existing user or a `sign_up` event in case of a new user.

> - **Trait key for the 'method' parameter in Login or Signup event**: Enter the trait key to pass the `method` parameter in `login` or `sign_up` event.

Reviewer: ```suggestion
- **Trait key for the 'method' parameter in Login or Signup event**: Enter the trait key to pass the `method` parameter in the `login` or `sign_up` event.
```

> - **Send 'generate_lead' event for 'identify' event**: Enable this setting to send a `generate_lead` event for a new user. Note that `generate_lead` will only be fired for new users and not for the existing ones. When enabled and a its new user, you also need to enter the following details:

Reviewer 2: ```suggestion
- **Send 'generate_lead' event for 'identify' event**: Enable this setting to send a `generate_lead` event for a new user. Note that `generate_lead` will only be fired for new users and not for the existing ones. When enabled and if it's a new user, you also need to enter the following details:
```

> - **Send 'generate_lead' event for 'identify' event**: Enable this setting to send a `generate_lead` event for a new user. Note that `generate_lead` will only be fired for new users and not for the existing ones. When enabled and its a new user, you also need to enter the following details:

Reviewer: Can we present this bit as an info block?

```
Note that `generate_lead` will only be fired for new users and not for the existing ones.
```

> - **Send 'generate_lead' event for 'identify' event**: Enable this setting to send a `generate_lead` event for a new user. Note that `generate_lead` will only be fired for new users and not for the existing ones. When enabled and its a new user, you also need to enter the following details:

Reviewer: ```suggestion
- **Send 'generate_lead' event for 'identify' event**: Enable this setting to send a `generate_lead` event for a new user. Note that `generate_lead` will only be fired for new users and not for the existing ones. When enabled in case of a new user, you also need to enter the following details:
```

Author: Actually, this means when both conditions are true:
- enabled
- new user

lmk if that makes sense

Reviewer: yeah - i had made the suggestion assuming the same. If you think the current sentence is more understandable, we can retain it.

>   - **Value - traits for lead generation**: Enter the value key to pass the `value` parameter for `generate_lead` event. 

Reviewer: ```suggestion
  - **Value - traits for lead generation**: Enter the value key to pass the `value` parameter for the `generate_lead` event. 
```

>   - **Currency - traits for lead generation**: Enter the currency key to pass the `currency` parameter for `generate_lead` event. 

Reviewer: ```suggestion
  - **Currency - traits for lead generation**: Enter the currency key to pass the `currency` parameter for the `generate_lead` event. 
```

<div class="infoBlock">

> Refer to <Link to="/destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode/#identify"><code class="inline-code">identify</code> call payload</Link> for more information.

Reviewer: ```suggestion
Refer to the <Link to="/destinations/analytics/google-analytics-4/google-analytics-4-cloud-mode/#identify"><code class="inline-code">identify</code> call payload</Link> for more information.
```
